### ISSUE TITLE
Regression in 7.18.0 causing "unsupported frontend protocol 1234.5679: server supports 2.0 to 3.0"

### ISSUE THREAD

**Issue by jan-tee** [brianc/node-postgres#2085]
I don't use pg directly but through sequelize (ORM). Updating to 7.17.0 works fine. 7.17.1 works fine, but the same unchanged code with pg 7.18.0 will throw an exception on connection:

`unsupported frontend protocol 1234.5679: server supports 2.0 to 3.0`

I don't have the full exception atm but this bit:

```
at Connection.connectingErrorHandler (C:\Users\jant\Repos\get-coffee\packages\get-coffee-api\node_modules\pg\lib\client.js:194:14)
    at Connection.emit (events.js:210:5)
    at Connection.EventEmitter.emit (domain.js:476:20)
    at TLSSocket.<anonymous> (C:\Users\jant\Repos\get-coffee\packages\get-coffee-api\node_modules\pg\lib\connection.js:144:12)
    at TLSSocket.emit (events.js:210:5)
    at TLSSocket.EventEmitter.emit (domain.js:476:20)
    at addChunk (_stream_readable.js:308:12)
    at readableAddChunk (_stream_readable.js:289:11)
    at TLSSocket.Readable.push (_stream_readable.js:223:10)
    at TLSWrap.onStreamRead (internal/stream_base_commons.js:182:23)
```

Why do I believe this is a regression: After downgrading to 7.17.1, connection works fine. After upgrading to 7.18.0, it breaks again.

**Comment by ricardo-trevizo-b** [comment]
Can confirm, I am experiencing the described issue.

I did try the workaround mentioned in https://github.com/petere/homebrew-postgresql/issues/51 and https://stackoverflow.com/questions/59190010/psycopg2-operationalerror-fatal-unsupported-frontend-protocol-1234-5679-serve
`PGGSSENCMODE=disable`

Alas to no avail. Downgrading to 7.17.1 works.

**Comment by harrisosserman** [comment]
Same happens for me as well.  I used to connect to my prod postgres db fine via SSL, but this update broke that.  I downgraded the version of this library until we find out what we should do.

**Comment by kapil-chandra** [comment]
Same here. Just spent 2 hours troubleshooting and saw that the upgrade from 7.17.1 to 7.18.0 caused this. I'm using Digital Ocean's managed postgres service which is using pg 11 and requires SSL.

**Comment by csuich2** [comment]
I appear to be hitting a similar problem, but with a different manifestation. Our CI server is now receiving an error when attempting to connect to our DB. I connected to the server and confirmed that it started using 7.18.0.

```
{ error: Invalid Username specified. Please check the Username and retry connection. The Username should be in <username@hostname> format.
    at Connection.parseE (/home/circleci/repo/common/temp/node_modules/.pnpm/registry.npmjs.org/pg/7.18.0_pg@7.18.0/node_modules/pg/lib/connection.js:624:13)
    at Connection.parseMessage (/home/circleci/repo/common/temp/node_modules/.pnpm/registry.npmjs.org/pg/7.18.0_pg@7.18.0/node_modules/pg/lib/connection.js:423:19)
    at TLSSocket.<anonymous> (/home/circleci/repo/common/temp/node_modules/.pnpm/registry.npmjs.org/pg/7.18.0_pg@7.18.0/node_modules/pg/lib/connection.js:139:22)
    at TLSSocket.emit (events.js:198:13)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at TLSSocket.Readable.push (_stream_readable.js:224:10)
    at TLSWrap.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
```

**Comment by ianwremmel** [comment]
Also seeing an issue with 7.18:

```
Error: Connection terminated unexpectedly
  File "/var/app/current/node_modules/pg/lib/client.js", line 255, col 9, in Connection.<anonymous>
    : new Error('Connection terminated unexpectedly')
  File "events.js", line 312, col 28, in Object.onceWrapper
  File "events.js", line 223, col 5, in Connection.emit
  File "domain.js", line 498, col 23, in Connection.EventEmitter.emit
  File "/var/app/current/node_modules/pg/lib/connection.js", line 149, col 10, in TLSSocket.<anonymous>
    self.emit('end')
  File "/var/app/current/node_modules/dd-trace/packages/dd-trace/src/scope/base.js", line 54, col 19, in null.<anonymous>
    return fn.apply(this, arguments)
  File "/var/app/current/node_modules/dd-trace/packages/dd-trace/src/scope/async_hooks.js", line 51, col 14, in Scope._activate
    return callback()
  File "/var/app/current/node_modules/dd-trace/packages/dd-trace/src/scope/base.js", line 12, col 19, in Scope.activate
    return this._activate(span, callback)
  File "/var/app/current/node_modules/dd-trace/packages/dd-trace/src/scope/base.js", line 53, col 20, in TLSSocket.bound
    return scope.activate(spanOrActive, () => {
  File "events.js", line 228, col 7, in TLSSocket.emit
  File "domain.js", line 498, col 23, in TLSSocket.EventEmitter.emit
  File "_stream_readable.js", line 1185, col 12, in endReadableNT
  File "internal/process/task_queues.js", line 81, col 21, in processTicksAndRejections
```

Could the new version be incompatible with datadog's tracer?

**Comment by eugene1g** [comment]
Related changes: https://github.com/brianc/node-postgres/pull/2082/files#diff-1449b1bce007eeb37fb3f3077d2dd1cbR120-R125
If those are disabled, connection works again. Related codes (`80877103` which translates to `1234. 5679) are explained at https://www.postgresql.org/docs/12/protocol-message-formats.html

**Comment by brianc** [comment]
ah crap! What version of postgres are y'all using?

**Comment by brianc** [comment]
I'll get a fix out asap

**Comment by ianwremmel** [comment]
10 something. The highest aurora serverless postgres version

**Comment by brianc** [comment]
k...I need to get a heroku app set up so I can try connecting to the postgres instance.  I have high on my list of things to do is to get some ssl enabled instances in docker so I can test them locally & in CI (though CI does use SSL AFAIK)...it'll take me a few min to get a test env set up w/ SSL

**Comment by brianc** [comment]
alright i can reproduce now. I'm really sorry about that! I'll have a fix out within the hour.  Literally the __next__ thing on my todo list is get SSL tests in better shape

**Comment by brianc** [comment]
also, in the mean time I'll not do any releases w/o manually testing against a heroku db w/ SSL.

**Comment by brianc** [comment]
found the problem, and tested a fix. figuring out a unit test for it & will push a new patch version w/ the fix. my deep apologies!

**Comment by brianc** [comment]
weirdly i didn't get notification on this issue until someone @ mentioned me.  I'm going to double-check my notification rules...I should have seen this 5 hours ago.  I specifically released in the morning to be on call if something went sideways like this. 😞

**Comment by brianc** [comment]
would one of y'all do me a quick favor & open a new issue on the repo to test the notifications? I tweaked my email rules.

**Comment by brianc** [comment]
> would one of y'all do me a quick favor & open a new issue on the repo to test the notifications? I tweaked my email rules.

thanks for doing that - got the email 😓

**Comment by brianc** [comment]
just waiting on travis then will push new semver patch

**Comment by brianc** [comment]
alrighty everyone...published `pg@7.18.1` 

sorry again for the disruption!  Please don't hesitate to open another issue if there's something amiss.  I'll be watching for the next several hours at least.  ❤️
